[Paddle] Add paddle frontend developer guide. (#13464)
openvinotoolkit/openvino: src/frontends/paddle/docs/operation_mapping_flow.md

> # OpenVINO Paddle Frontend operator enabling Flow

Reviewer: Most of diagrams contain outdated component names: nGraph, CNNNetwork. I think we definitely need to avoid it.

Reviewer 2: Yes, it is why I asked to update all diagrams

Author: updated with mermaid.

> # OpenVINO Paddle Frontend operator enabling Flow

Reviewer 3: ```suggestion
# OpenVINO Paddle Frontend Operator Enabling Flow
```

1. Declare `CreatorFunction` for the Paddle operator and register it to the map in `src/op_table.cpp`.
   * The map is retrived from:

     https://github.com/openvinotoolkit/openvino/blob/7d5e0abcaa03703de9918ece2115e6ea652c39e0/src/frontends/paddle/src/op_table.cpp#L106

> 2. Implement the operator mapper `CreatorFunction` in path `src/op/`.

Reviewer 2: I am not sure that I am as an external user will be able to implement the operator without the example.

Author: updated

> 2. Implement the operator mapper `CreatorFunction` in path `src/op/`.

Reviewer 3: ```suggestion
2. Implement the operator mapper `CreatorFunction` in the `src/op/` path.
```

> 3. Add unit-tests [OpenVINO™ Paddle Frontend unit test README](tests.md)

Reviewer 3: ```suggestion
3. Add unit-tests. For more information, refer to the [OpenVINO™ Paddle Frontend unit-tests readme](tests.md)
```

> 3. Add unit-tests * [OpenVINO™ Paddle Frontend unit test README](tests.md)

Reviewer 3: why * after unit-tests?

Author: fixed.

# OpenVINO Paddle Frontend operator enabling Flow

1. Declare `CreatorFunction` for the Paddle operator and register it to the map in `src/op_table.cpp`.
   * The map is retrived from:
     ```cpp

>      std::map<std::string, CreatorFunction> get_supported_ops() {

Reviewer 2: I think will be better to add a real code to such place, please take a look: https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/ir/docs/support_new_opset.md
Here I insert a part of the existed file in the md document.

Author: done

         return {{"arg_max", op::argmax},
                 ......
     ```
2. Implement the operator mapper `CreatorFunction` in path `src/op/`.
3. Add unit-tests * [OpenVINO™ Paddle Frontend unit test README](tests.md)

## See also
* [OpenVINO™ Paddle Frontend README](../README.md)

> * [OpenVINO™ Frontend README](TODO)

Reviewer 2: I think this link also exists

Author: done
